### ISSUE TITLE
Separately Formatting Tags in Environments Versus References (`tagformat`)

### ISSUE THREAD

**Issue by geometrian** [mathjax/MathJax#2736]
# Is your feature request related to a problem? Please describe.

The [`tagformat` option](https://docs.mathjax.org/en/latest/input/tex/extensions/tagformat.html) is really great as far as it goes, but (as far as I can tell) it's missing a way to format the reference and the referent separately.

To clarify: the particular use-case would be something like having the equation number *inside* the `\begin{equation}`/`\end{equation}` block be "(1)", but the equation *reference* `\eqref{...}` be "Equation 1" (the whole thing linked to the equation). That is, the following source:
```html
\begin{equation}
    f(x) = x^2
    \label{eq:quadratic}
\end{equation}

<p>A simple quadratic function is given in \eqref{eq:quadratic}.</p>
```
Would render something like:
```text
    f(x) = x^2        (1)

A simple quadratic function is given in Equation 1.
```
This evokes established LaTeX style in some fields.

However, as it is, the configured function `MathJax.tex.tagformat.tag` is used for both places, making this apparently impossible.

# Describe the solution you'd like

Put simply, I would like to see a way to do this added into MathJax.

I don't want to say how this *must* be done, but *perhaps* the way it might be exposed would be to make another function (`reftag`? `eqref`?) inside `tagformat` that by default calls `tag`. This solution has the advantage of preserving backward-compatibility.

The config for the above use-case would then be something like:
```js
MathJax = {
    loader: { load: [ "[tex]/tagformat" ] },
    tex: {
        //...
        tags: "ams",
        packages: { "[+]": [ "tagformat" ] },
        tagformat: {
            tag:    (tag) => "("+tag+")",
            reftag: (tag) => "Equation "+tag,
        }
    }
    //...
};
```

# Describe alternatives you've considered

Not really applicable; the alternative is to not support this feature. As before, that would be a shame since certain fields require this formatting.

If this feature is already supported, then it should be documented.

# Additional context

Not applicable.

**Comment by dpvc** [comment]
You are right, there is no mechanism for that.  But you can do something like

```
\renewcommand{\eqref}[1]{Equation \ref{#1}}
```

in order to get `Equation 1` for the equation tagged with `(1)`, where the `1` is the link.  Perhaps that is sufficient for now?

**Comment by imallett** [comment]
Adding `eqref: ["{\\text{Equation }\\ref{#1}}",1]` to `MathJax.tex.macros` indeed gets me very nearly what I want for now (not quite though—I'd like the whole "Equation 1" to be a link, not just the number, though I don't think this is possible even in LaTeX without more stuff).

**Comment by dpvc** [comment]
I've made a pull request to implement this in v4.

**Comment by dpvc** [comment]
Added in v4.0.
